### DISCUSSION TITLE
Display relational (translation) field

### DISCUSSION THREAD

**Discussion by superhp** [directus/directus#10867] [Archive]
Lets say there are three collections:
1. Translations (it contains translations for category names)
2. Categories (the name of category comes from relation with translations)
2. Shops (one shop can belong to one category so it has CategoryId)

How can I display category name when I open Shops collection? If I set Display Template to take Category -> Translation -> Name, it shows "--" instead. For Category to show it's name, we can choose "Translations" display option instead of "Related Values". But in this case, it is only possible to choose "Related Values". And same issue is with Interface template. 

Do you have any ideas how this could be solved?

**Comment by rijkvanzanten** [comment]
> Do you have any ideas how this could be solved?

Could you try configuring the related-values display for `categories` on `shops`, and then chose the top-level `translations` field (so not the nested `name`). Then on the `translations` field itself on the `categories` collection, set the display to `translations`. This _should_ allow you to pull in the translations display from the nested Translations table onto the field listed in the related-values display. 🤔

**Reply by superhp** [reply]
Yes, it works now! So, instead of Translation -> Name, I just needed Translation on display template of Category field in Shops collection. Thank you! 
There is also one strange thing. Let's say, I actually have Shops - Shops Categories - Categories (M2M, shop can belong to many categories). In Shops collection, there are three columns: 
![image](https://user-images.githubusercontent.com/7535630/149121856-f2baa795-7d40-40df-8b0f-5eefceb04c45.png)
As you can see, the third column has a number of categories that the shop belongs to and if you click on it, the list of categories expands. The issue is  with the category names there. If I set Display template as Translation or Translation -> Name for this field, it only shows "--" instead of actual names. But it somehow works, when I set a combination of Translation Translation -> Name. 
![image](https://user-images.githubusercontent.com/7535630/149122680-ba77baeb-2e7c-4e11-b0f6-32b76921e13f.png)
Then it shows "Name--" as it is in the previous screenshot. This is not an issue for me, I don't think I even need this column. But I think this is some kind of misbehavior.

**Reply by rijkvanzanten** [reply]
I'm not quite following the exact schema here, so I'll have to play around with reproducing your exact setup to confirm that. In any case, I think the `--` problem you ran into before came from the fact that you were trying to render a list of values as a single value point, like `Translations > Name` would result in `["Naam", "Name", "Nom"]`, which can't be rendered in that single slot in the display template, causing the `--` 🤔

**Reply by manufest** [reply]
@rijkvanzanten 
I have the same issue with a nested recursive o2m relation with translations using the tree view interface. I have to set Display Template to [Translations] [Translations -> Label] to show the actual translation in the edit form followed by --. Any other combination or Translations alone won't work and I get --. This only occurs in interface options, display options with [Translations] set is working correctly. Running latest 9.12.1

![Bildschirmfoto 2022-06-04 um 11 52 04](https://user-images.githubusercontent.com/13063722/171994524-73520f37-4c0a-4fd1-9ed0-c922268b6891.png)
![Bildschirmfoto 2022-06-04 um 11 52 51](https://user-images.githubusercontent.com/13063722/171994539-63749620-f04e-4050-860d-35629d32ec59.png)

**Comment by svitzthum** [comment]
I think I have the exact same problem. Displaying translations of related entities doesn't work, neither in the edit page nor the item list.

My configuration is the same as described by @superhp 

I have a `category` collection with a translated `name`
![image](https://user-images.githubusercontent.com/29091217/153900406-cbd9e018-e2b2-49cb-9fe2-c54cee9bee16.png)
![image](https://user-images.githubusercontent.com/29091217/153901524-cc0926ff-6eab-4ecb-96af-71f6da083c61.png)

I have a `shop` collection using a m2m relation to reference `categories`
![image](https://user-images.githubusercontent.com/29091217/153901066-b079c460-336c-4b8f-b095-2b3fe4370c48.png)
![image](https://user-images.githubusercontent.com/29091217/153901227-3786869c-7738-41a7-b468-bda3342b7469.png)

This is how it is shown if I follow the selected answer (I hope I understood it correctly 😅)
![image](https://user-images.githubusercontent.com/29091217/153902266-84c62d4a-c6c8-4e9e-9f87-f41d0a2620e2.png)
![image](https://user-images.githubusercontent.com/29091217/153902429-9a7491e4-82c3-4fcd-bb13-9581bbd197f9.png)

@rijkvanzanten do you have any additional ideas on this topic?
Thank you very much 😃

**Comment by superhp** [comment]
@rijkvanzanten It seems the functionality was still working with 9.5.0 version but it was broken with 9.5.1 (I didn't have much time to investigate but guess it was broken by fixing https://github.com/directus/directus/issues/9358, https://github.com/directus/directus/pull/9421, CC @Nitwel). And it is still broken with 9.5.2. After the upgrade to 9.5.1 or 9.5.2 the relational translation fields become just empty as in my original post or as in @semmel87 example.

**Reply by svitzthum** [reply]
Thank you for this information. 👍 
Indeed we are using 9.5.1

**Reply by HemboPagi** [reply]
Hello, i am using 9.5.2 and I can confirm it is not working

**Comment by yassilah** [comment]
Hi! I can confirm the issue persists with 9.6.0 but works with 9.5.0. I'm new to directus so I'm not sure where to look but i'll try to find where the regression is coming from

**Comment by yassilah** [comment]
Ok, I actually just saw that it was merely a CSS issue, everything works as expected, the value is just hidden, i'll make a PR :)

**Reply by yassilah** [reply]
Should be fixed with https://github.com/directus/directus/pull/12153 ✌️
